light_curves/lc_classifier.ipynb

```diff
@@ -353,82 +419,80 @@
-    "            \n",
-    "        #for each band\n",
-    "        for bandid in range(len(full_bandname)):\n",
+    "    #make a dictionary to hold the light curve\n",
+    "    zerosingle = {\"objectid\": oid, \"label\": label, \"band\": band, \"time\": timelist, \n",
+    "                  \"flux\": np.zeros(len(timelist)), \"err\":np.zeros(len(timelist))}\n",
     "    \n",
-    "            #set up to only have time as an index\n",
-    "            single_obj = df_lc.loc[oid,:,full_bandname[bandid],:].reset_index().set_index('datetime')\n",
-    "    \n",
-    "            #sort the time index\n",
-    "            single_obj = single_obj.sort_index()\n",
+    "    return zerosingle\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true,
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "#for the case where there is no photometry in a band:\n",
+    "#make light curves with flux, err set to zeros\n",
+    "\n",
+    "#what is the full set of unique band names?\n",
+    "full_bandname = df_lc.band.unique()\n",
+    "\n",
+    "#setup a list to store empty light curves\n",
+    "zerosingle_list = [] \n",
+    "\n",
+    "#for each object in each band\n",
+    "for oid , singleoid in df_lc.groupby(\"objectid\"):\n",
+    "                                  \n",
+    "    #this is the list of bandnames for that object                                \n",
+    "    oid_bandname = singleoid.band.unique()\n",
     "    \n",
-    "            #change the time index to have values in fullindex, and use 'nearest' to fill\n",
-    "            #in values of the new time indices\n",
-    "            #this is where the real work gets done\n",
-    "            single_reindex = single_obj.reindex(fullindex, method = 'nearest')\n",
+    "    #figure out which bands are missing\n",
+    "    missing = list(set(full_bandname).difference(oid_bandname))\n",
     "    \n",
-    "            #previous command returns an unnamed index, so name it\n",
-    "            single_reindex.index.rename('datetime', inplace= True)\n",
+    "    #if it is not the complete list, ie some bandnames are missing:                            \n",
+    "    if len(missing) > 0:\n",
     "    \n",
-    "            #then we really want a multiindex df, so first remove indices, then set the ones we want\n",
-    "            single_reindex.reset_index(inplace= True)\n",
-    "            \n",
-    "            #put oid and bandname back into the dataframe\n",
-    "            single_reindex['objectid'] = oid\n",
-    "            single_reindex['band'] = full_bandname[bandid]\n",
+    "        #make new dataframe for this object with zero flux and err values\n",
+    "        for m in range(len(missing)):\n",
+    "            label = str(singleoid.label.unique().squeeze())\n",
+    "            zerosingle = make_zero_lc(oid, missing[m], label)\n",
```

Comment:
Can you iterate over the band name instead of the index?

```suggestion
    "        for band in missing:\n",
    "            label = str(singleoid.label.unique().squeeze())\n",
    "            zerosingle = make_zero_lc(oid, band, label)\n",
```